docs/sdks/js-ts/overview.mdx

```diff
@@ -60,139 +63,166 @@ In Kwil, accounts are identified by a public key. Kwil supports Secp256k1 Public
-const publicKey = await Utils.recoverSecp256k1PubKey(signer, 'Welcome to our app! Sign this message to reveal your public key.'); 
+const publicKey = await Utils.recoverSecp256k1PubKey(
+  signer,
+  "Welcome to our app! Sign this message to reveal your public key."
+);
 ```
 
 ### Database Identifiers (DBID)
 
 In Kwil, databases are identified by a 'database identifier', which is a hex encoded SHA224 Hash of the database name and public key, prepended with an `x`.
 
 ```javascript
-const dbid = kwil.getDBID('public_key', 'database_name')
+const dbid = kwil.getDBID("public_key", "database_name");
 ```
 
-## Database Info
+## Signers
 
-### List Databases 
+Certain operations in Kwil require signature authentication from the user (e.g. deploy database, drop database, execute CUD actions, certain read operations, etc).
 
-``` javascript
-const res = await kwil.listDatabases("public_key")
-// res.data = ["db1", "db2", "db3"]
-```
+To manage signing, Kwil-JS uses a `KwilSigner` class. Out of the box, Kwil-Js supports signers from [EthersJS](https://github.com/ethers-io/ethers.js) (v5 and v6). You can also pass a signing callback function (see [below](#custom-signers)).
 
-### Read a Deployed Database Schema
-``` javascript
-const dbid = kwil.getDBID("public_key", "database_name")
-const schema = await kwil.getSchema(dbid)
+```javascript
+import { Utils, KwilSigner } from "@kwilteam/kwil-js";
+import { BrowserProvider } from "ethers";
 
-/*
-    schema.data = {
-        owner: Uint8Array,
-        name: "database_name",
-        tables: [ tableObject1, tableObject2, tableObject3 ],
-        actions: [ action1, action2, action3 ],
-        extensions: [ extension1, extension2 ]
-    }
-*/
+// get ethers signer
+const provider = new BrowserProvider(window.ethereum);
+const signer = await provider.getSigner();
+
+// get secp256k1 public key
+const publicKey = await Utils.recoverSecp256k1PubKey(signer);
+
+// create kwil signer
+const kwilSigner = new KwilSigner(signer, publicKey);
 ```
 
-### Get Account Info
+#### Custom Signers
 
-``` javascript
-const res = await kwil.getAccount("public_key")
+If you wish to sign with something other than an EtherJS signer, you may pass a callback function that accepts and returns a `Uint8Array()` and the enumerator for the signature type used.
 
-/*
-    res.data = {
-        address: "public_key",
-        balance: "some_balance",
-        nonce: "some_nonce"
-    }
-*/
+Currently, Kwil supports two signature types:
+
+| Type      |   Enumerator   |
+| :-------- | :------------: |
+| Secp256k1 | 'secp256k1_ep' |
+| Ed25519   |   'ed25519'    |
+
+To use an ed25519 signature:
+
+```javascript
+import nacl from "tweetnacl";
+import { KwilSigner } from "@kwilteam/kwil-js";
+
+// create keypair and signer
+const keys = nacl.sign.keyPair();
+const customSigner = (msg) => nacl.sign.detached(msg, keys.secretKey);
+
+const kwilSigner = new KwilSigner(customSigner, keys.publicKey, "ed25519");
```

Comment:
I'd rather just leave the ed25519 example out.  Rather, describing what each ones means might be more concise.

Reply:
Do you think the ed25519 example is useful as a practical example as a custom signer?

Reply:
Eh I guess.  Still explain the others tho.  It is not clear what `secp256k1_ep` means.